.travis.yml

```diff
@@ -29,16 +29,17 @@ before_install:
     - source venv/bin/activate
     - python --version  # just to check
     - pip install -U pip wheel # upgrade to latest pip find 3.5 wheels; wheel to avoid errors
-    - retry pip install pytest pytest-cov pytest-dependency flake8 coverage codecov chardet "setuptools!=47.2.0" docutils check-manifest
```

Comment:
We do use flake8 -- see below the call to `flake8`. In theory we should add a Makefile entry to `Makefile` that runs the steps below. Then `script:` could just be `make test`, where `make test` is an alias for the commands below. So I do think `flake8` should be part of the `requirements_dev` or whatever...

Reply:
Okay I've added make check (rather than test), as the phony was already configured and it's what I've used before, plus more Google hits. I could alias test to check if you'd like?

It runs lots of stuff as you'll see.

Technically the Python dictionary checks get run twice via the basic pytest and the check-dictionaries one, unless we swap to calling sort and trim individually and keeping them in step, there's no easy way round that. I think if you've got a failure it will bomb out anyway.